ompi/mca/coll/shared/coll_shared_reduce.c

```diff
@@ -9,7 +9,7 @@ int mca_coll_shared_reduce_intra(const void *sbuf, void* rbuf, int count,
     ptrdiff_t extent, lower_bound;
     ompi_datatype_get_extent(dtype, &lower_bound, &extent);
     if (count*extent <= 256) {
-        mca_coll_shared_reduce_binomial(sbuf, rbuf, count, dtype, op, root, comm, module);
+        mca_coll_shared_reduce_linear(sbuf, rbuf, count, dtype, op, root, comm, module);
```

Comment:
Why do we force the linear for small messages ? Logically the binomial seems like a better choice.